packages/components/src/toggle-group-control/toggle-group-control/component.tsx

```diff
@@ -22,6 +20,54 @@ import { VisualLabelWrapper } from './styles';
 import * as styles from './styles';
 import { ToggleGroupControlAsRadioGroup } from './as-radio-group';
 import { ToggleGroupControlAsButtonGroup } from './as-button-group';
+import { useTrackElementOffsetRect } from '../../utils/element-rect';
+import { useOnValueUpdate } from '../../utils/hooks/use-on-value-update';
+import { useEvent, useMergeRefs } from '@wordpress/compose';
+
+function useSubelementAnimation(
```

Comment:
This utility can be quite difficult to understand for someone who is not familiar with your work and the component — we should add some code comments to help give more context

Reply:
agreed, hard to understand in its current state, this is gonna get a fast follow that takes it to the utils dir, and then another that makes Tabs use it too, so in that process it can be properly documented. I have those changes already locally.

Do you think it's okay to merge as is now with this context, since your concern will be addressed very soon?

Reply:
It doesn't hurt to add comments now, while you're also adding the ones for the CSS border radius calculation. Good to know that there are plans to move the code to a util

Reply:
Fair, done.